src/main/java/org/openrewrite/analysis/search/FindFlowBetweenMethods.java

```diff
@@ -115,32 +117,44 @@ private boolean conditional(InvocationMatcher sinkMatcher, Cursor cursor) {
                 }
             }
 
-            private LocalFlowSpec<Expression, Expression> getFlowSpec(Expression source, MethodMatcher sink) {
+            private LocalFlowSpec<Expression, Expression> getFlowSpec(Cursor srcCursor, MethodMatcher sink) {
                 final InvocationMatcher sinkMatcher = InvocationMatcher.fromMethodMatcher(sink);
                 switch (flow) {
                     case "Data":
                         return new LocalFlowSpec<Expression, Expression>() {
 
                             @Override
-                            public boolean isSource(Expression expression, Cursor cursor) {
-                                return expression == source;
+                            public boolean isSource(DataFlowNode srcNode) {
+                                return srcNode
+                                        .asExpr()
+                                        .map(src -> Expr.viewOf(srcCursor)
```

Comment:
This is inside two separate instantiations of `LocalFlowSpec`. So, instead of computing it once at the top of the method, we could make this a separate private method call which both instantiations can use.